core/conversion/converters/impl/plugins/checkshape_plugin.cpp

```diff
@@ -0,0 +1,425 @@
+  outputDims.nbDims = 1;
+  outputDims.d[0] = exprBuilder.constant(1); // shape of shape
+
+#ifndef NDEBUG
+  std::cout << "  outputIndex is " << outputIndex << ", nbDims=" << outputDims.nbDims << std::endl;
+#endif
+
+  return outputDims;
+}
+
+inline bool CheckShapePlugin::supportsFormatCombination(
+    int32_t pos,
+    const PluginTensorDesc* inOut,
+    int32_t nbInputs,
+    int32_t nbOutputs) {
+#ifndef NDEBUG
+  std::cout << "CheckShapePlugin::supportsFormatCombination(): this=" << this << std::endl;
+#endif
+
+  assert(
+      pos < (nbInputs + nbOutputs) && nbInputs == CHECKSHAPE_PLUGIN_NUM_INPUT &&
+      nbOutputs == CHECKSHAPE_PLUGIN_NUM_OUTPUT);
+
+  switch (pos) {
+    case 0: // input0
+      return inOut[0].type == DataType::kINT32 && inOut[0].format == PluginFormat::kLINEAR;
+    case 1: // input1
+      return inOut[1].type == DataType::kINT32 && inOut[1].format == PluginFormat::kLINEAR;
+    case 2: // output0
+      return (inOut[2].type == inOut[0].type) && inOut[2].format == PluginFormat::kLINEAR;
+    default:
+      return false;
+  }
+}
+
+inline void CheckShapePlugin::configurePlugin(
+    const DynamicPluginTensorDesc* in,
+    int32_t nbInputs,
+    const DynamicPluginTensorDesc* out,
+    int32_t nbOutputs) {
+#ifndef NDEBUG
+  std::cout << "CheckShapePlugin::configurePlugin(): this=" << this << std::endl;
+#endif
+  assert(nbInputs == CHECKSHAPE_PLUGIN_NUM_INPUT);
+  assert(nbOutputs == CHECKSHAPE_PLUGIN_NUM_OUTPUT);
+  assert(in[0].desc.type == DataType::kINT32);
+  assert(in[0].desc.format == PluginFormat::kLINEAR);
+  assert(in[0].desc.dims.nbDims == 1);
+  assert(in[1].desc.type == DataType::kINT32);
+  assert(in[1].desc.format == PluginFormat::kLINEAR);
+  assert(in[1].desc.dims.nbDims == 1); // [batchsize, channel]
+  assert(inputRank == in[0].desc.dims.d[0]);
+}
+
+inline size_t CheckShapePlugin::getWorkspaceSize(
+    const PluginTensorDesc* inputs,
+    int32_t nbInputs,
+    const PluginTensorDesc* outputs,
+    int32_t nbOutputs) const {
+#ifndef NDEBUG
+  std::cout << "CheckShapePlugin::getWorkspaceSize(): this=" << this << std::endl;
+#endif
+  return 0;
+}
+
+inline int32_t CheckShapePlugin::enqueue(
+    const PluginTensorDesc* inputDesc,
+    const PluginTensorDesc* outputDesc,
+    const void* const* inputs,
+    void* const* outputs,
+    void* workspace,
+    cudaStream_t stream) {
+#ifndef NDEBUG
+  std::cout << "CheckShapePlugin::enqueue(): this=" << this << std::endl;
+#endif
+  int status = 0;
+
+  int32_t* inShape = (int32_t*)inputs[0];
+  int32_t* expandShape = (int32_t*)inputs[1];
+  int32_t* output = (int32_t*)outputs[0];
+
+  int* h_inShape = (int*)malloc(sizeof(int) * inputRank);
+  int* h_expandShape = (int*)malloc(sizeof(int) * expandRank);
+  cudaMemcpy(h_inShape, inShape, sizeof(int) * inputRank, cudaMemcpyDeviceToHost);
+  cudaMemcpy(h_expandShape, expandShape, sizeof(int) * expandRank, cudaMemcpyDeviceToHost);
+
+  int* h_output = (int*)malloc(sizeof(int) * 1);
+  *h_output = 0;
```

Comment:
The sole output is 0? Is the role of this plugin to fail if the checks below dont pass? can you write some comments on what this is supposed to do?